src/AppInstallerRepositoryCore/Public/winget/Checkpoint.h

```diff
@@ -53,6 +54,13 @@ namespace AppInstaller::Checkpoints
             m_checkpointDatabase->SetDataValue(m_checkpointId, dataType, fieldName, values);
         }
 
+        // Update a single existing field value for a data type.
+        void Update(T dataType, const std::string& fieldName, const std::string& value)
+        {
+            m_checkpointDatabase->RemoveDataType(m_checkpointId, dataType);
+            m_checkpointDatabase->SetDataValue(m_checkpointId, dataType, fieldName, { value });
```

Comment:
I'm fairly sure that you could fix this with an [upsert](https://sqlite.org/lang_upsert.html), but if that isn't to be done then a transaction should be used to preserve the atomic nature of updating the value.

Reply:
I found this old comment that mentioned upsert: https://github.com/microsoft/winget-cli/blob/cd8e9920bcd5cc28bf9cb1217a82847352da0e09/src/AppInstallerRepositoryCore/Microsoft/Schema/1_1/ManifestMetadataTable.cpp#L94C1-L96

I'm assuming this comment still applies so I went ahead and used a transaction for updating the value.